components/engine_traits/src/import.rs

```diff
@@ -20,6 +20,13 @@ pub trait ImportExt: CFHandleExt {
         files: &[&str],
     ) -> Result<()>;
 
+    fn ingest_external_file_optimized(
```

Comment:
`ingest_external_file_optimized` is more like a kind of implementation of `ingest_external_file_cf`

Reply:
Understood. I'm reluctant to merge the two here since that would introduce a change in behavior. I'm just mirroring what is already implemented. If there's existing code duplication let's fix it elsewhere.